flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecTableSourceScan.java

```diff
@@ -133,6 +139,21 @@ protected Transformation<RowData> translateToPlanInternal(
                             outputTypeInfo,
                             sourceParallelism,
                             sourceParallelismConfigured);
+
+            LineageDataset tableLineageDataset =
```

Comment:
Here I think we need to check if the source implements `LineageVertexProvider` first. If so we extract datasets from it then we add additional table layer information to it and construct the `TableLineageDataset`.


My understanding is that without information from connector, it is not possible to build a `LineageVertex` purely from tables. From the table layer we can only tell the type of the external system by checking `connector` option, like Kafka, JDBC and so forth, but we can't identify which Kafka cluster / database instance the lineage dataset belongs to. 


This is also the reason I'm confused by the `namespace` in `TableLineageDatasetImpl`. Table planner can only add additional info such as schema and object identifier, but the root lineage dataset should be provided by connectors.


The same issue for `CommonExecSink`.

Reply:
Thanks for the idea proposed. Yes, I think read the lineage info from connector totally makes sense. If the connector doesn't implement LineageVertexProvider, then we will leave namespace as empty here.